File: PracticeProject/app/src/main/java/com/zero/practiceproject/MainActivity.java
package com.zero.practiceproject;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    Button firstActivty;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        firstActivty = findViewById(R.id.first_activity);
    }

    public void onClick(View view){
        switch (view.getId()){
            case R.id.first_activity:
                Intent intent = new Intent(MainActivity.this, FirstActivity.class);
                startActivity(intent);
                break;
            default:
                break;
        }
    }


}

File: MotionLayout/example/src/main/java/com/example/example/MainActivity.kt
package com.example.example

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClick(v: View) {
        val target = "com.example.example.${(v as Button).text}"
        startActivity(Intent(this, Class.forName(target)));
    }
}